Document `make backport` details and usage. (#580)
submariner-io/submariner-website: src/content/development/backports/_index.en.md

### Standalone pull requests

> Backporting a commit is automated by running `make backport release=<release-branch> pr=<PR to cherry-pick>`. The `make` target runs a script,
> [backport.sh](https://github.com/submariner-io/shipyard/scripts/shared/backport.sh). The script is originally being used by Kubernetes community
> and has been reused, and edited to suit the needs, from [their repo](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh).
> The script does the following:

Reviewer: ```suggestion
Backporting a commit is automated by running:

`make backport release=<release-branch> pr=<PR number to backport>`

The make target runs [`backport.sh`](https://github.com/submariner-io/shipyard/scripts/shared/backport.sh, originally developed by the Kubernetes community as [`cherry_pick_pull.sh`](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh).

The script does the following:
```

> 1. Creates a PR on `<release-branch>` with title `Automated cherry pick of <original PR number> <original PR title>`.
> 2. Adds `backport-handled` label to the original PR.
> 3. Gives an option to create patches to a release branch without making a PR by setting `DRY_RUN` variable.
>    When `DRY_RUN` is set the script will leave you in a branch containing the commits you cherry-picked.
> 4. Multiple PRs can be cherry-picked together by passing the comma-separated list of PRs to be cherry-picked like `pr=<PR1,PR2>`.   

Reviewer 2: ```suggestion
1. Cherry-picks the commits from the PR onto <remote branch>.
2. Creates a PR on `<release-branch>` with the title `Automated backport of <original PR number>: <original PR title>`.
3. Adds the `backport-handled` label to the original PR.

The DRY_RUN environment variable can be set to skip creating the PR. When set, it leaves you in a branch containing the commits that were cherry-picked.

Multiple PRs can be backported together by passing a comma-separated list of PR numbers, eg `pr=630,631`.   
```

> The script assumes following values. Please change them according to your setup.
> 1. Remote for the upstream repository, `UPSTREAM_REMOTE`, is set to `origin`.
> 2. Remote for your forked repository, `FORK_REMOTE`  is set to `GITHUB_USER`.

Reviewer 2: ```suggestion
The script uses the following environment variables. Please change them according to your setup.
* `UPSTREAM_REMOTE`: the remote for the upstream repository. Defaults to `origin`.
* `FORK_REMOTE`: the remote for your forked repository. Defaults to `GITHUB_USER`.
* `GITHUB_USER`: needs to be set to your GitHub username.
```

> 3. `GITHUB_USER` needs to be set to your `GitHub username`.

Reviewer: ```suggestion
3. `GITHUB_USER` needs to be set to your GitHub username.
```

### Standalone pull requests

> Backporting a commit is automated by running:

Reviewer 2: ```suggestion
Backporting a pull request (PR) is automated by running:
```

`make backport release=<release-branch> pr=<PR to cherry-pick>`

The `make` target runs a script,
[backport.sh](https://github.com/submariner-io/shipyard/scripts/shared/backport.sh), originally being used by Kubernetes community

> and has been reused, and edited to suit the needs, from [their repo](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh).

Reviewer 2: ```suggestion
[backport.sh](https://github.com/submariner-io/shipyard/scripts/shared/backport.sh), originally developed by the [Kubernetes community](https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh).
```

The script does the following:

> 1. Cherry-picks the commits from the PR onto <remote branch>.

Reviewer 2: ```suggestion
1. Cherry-picks the commits from the PR onto `<remote branch>`.
```

2. Creates a PR on `<release-branch>` with the title `Automated backport of <original PR number>: <original PR title>`.

> 3. Adds the `backport-handled` label to the original PR and `automated-backport` label to the backported PR.

Reviewer 2: ```suggestion
3. Adds the `backport-handled` label to the original PR and the `automated-backport` label to the backported PR.
```
